src/uu/sort/src/external_sort/mod.rs

```diff
@@ -91,205 +62,150 @@ where
-                        minimum_buffer_size
-                    } else {
-                        adjusted_buffer_size
-                    };
-                chunk.push(seq);
-
-                if total_read >= adjusted_buffer_size {
-                    super::sort_by(&mut chunk, &self.settings);
-                    self.write_chunk(
-                        &iter.tmp_dir.path().join(iter.chunks.to_string()),
-                        &mut chunk,
-                    )?;
-                    chunk.clear();
-                    total_read = 0;
-                    iter.chunks += 1;
-                }
-            }
-            // write the last chunk
-            if chunk.len() > 0 {
-                super::sort_by(&mut chunk, &self.settings);
-                self.write_chunk(
-                    &iter.tmp_dir.path().join(iter.chunks.to_string()),
-                    &mut chunk,
-                )?;
-                iter.chunks += 1;
-            }
-
-            // initialize buffers for each chunk
-            //
-            // Having a right sized buffer for each chunk for smallish values seems silly to me?
-            //
-            // We will have to have the entire iter in memory sometime right?
-            // Set minimum to the size of the writer buffer, ~8K
-            //
-            const MINIMUM_READBACK_BUFFER: u64 = 8200;
-            let right_sized_buffer = adjusted_buffer_size
-                .checked_div(iter.chunks)
-                .unwrap_or(adjusted_buffer_size);
-            iter.max_per_chunk = if right_sized_buffer > MINIMUM_READBACK_BUFFER {
-                right_sized_buffer
-            } else {
-                MINIMUM_READBACK_BUFFER
-            };
-            iter.buffers = vec![VecDeque::new(); iter.chunks as usize];
-            iter.chunk_offsets = vec![0 as u64; iter.chunks as usize];
-            for chunk_num in 0..iter.chunks {
-                let offset = fill_buff(
-                    &mut iter.buffers[chunk_num as usize],
-                    File::open(iter.tmp_dir.path().join(chunk_num.to_string()))?,
-                    iter.max_per_chunk,
-                )?;
-                iter.chunk_offsets[chunk_num as usize] = offset;
-            }
-        }
+            &iter.tmp_dir.path().join(iter.chunks.to_string()),
+            &mut chunk,
+        );
+        iter.chunks += 1;
+    }
 
-        Ok(iter)
+    // initialize buffers for each chunk
+    //
+    // Having a right sized buffer for each chunk for smallish values seems silly to me?
+    //
+    // We will have to have the entire iter in memory sometime right?
+    // Set minimum to the size of the writer buffer, ~8K
+
+    const MINIMUM_READBACK_BUFFER: usize = 8200;
+    let right_sized_buffer = settings
+        .buffer_size
+        .checked_div(iter.chunks)
+        .unwrap_or(settings.buffer_size);
+    iter.max_per_chunk = if right_sized_buffer > MINIMUM_READBACK_BUFFER {
+        right_sized_buffer
```

Comment:
would it be possible to add a test for this case?

Reply:
done...